patches/net/minecraft/world/item/CreativeModeTab.java.patch

```diff
@@ -230,6 +230,20 @@
 +            java.util.stream.Stream.of(tabs).map(net.minecraft.resources.ResourceKey::location).forEach(this.tabsAfter::add);
 +            return this;
 +        }
++
++        /**
++         * Helper to set this tabs contents to everything in the supplied {@link net.neoforged.neoforge.registries.DeferredRegister}.
++         * Due to the {@link ItemLike} generic, you can pass Item or Block DeferredRegisters into this by default.
++         * Entries added through this method are filtered out if {@link ItemLike#asItem()} returns {@link Items.AIR}, to account for blocks that do not have items.
++         * Note that like the vanilla method this overrides any other calls for this tab to {@link #displayItems}.
++         */
++        public CreativeModeTab.Builder displayItems(net.neoforged.neoforge.registries.DeferredRegister<? extends ItemLike> register) {
```

Comment:
We should not be binding DR into vanilla systems like this. If you want an easier method of supplying this data to `CreativeModeTab$Builder`, it should take an output of a DR, but not a DR directly.

You should be able to have it take a `Collection<Holder<? extends ItemLike>>` or a similar container.

Reply:
Though, unfortunately, due to the return signature of `getEntries`, it will need to be
`Collection<? extends Holder<? extends ItemLike>> entries`

Reply:
I don't like the idea of not having DR there directly and that generic is kinda ugly (makes it a lot less obvious you can easily pass a DR in), but I can understand not wanting to bind DR to vanilla like this.

Reply:
You can _suggest_ using `DeferredRegister#getEntries()` in the javadocs for the method, since it should be one of (if not the most) common cases.